core/indexer/common.go

```diff
@@ -367,62 +367,26 @@ func serializeAccountsHistory(accounts map[string]*AccountBalanceHistory) []byte
 func prepareSerializedDataForATransaction(
 	tx *Transaction,
 	selfShardID uint32,
-	isMBOfTxInDB bool,
+	_ bool,
 ) (meta []byte, serializedData []byte) {
-	var err error
-	if isMBOfTxInDB {
-		if !isCrossShardDstMe(tx, selfShardID) || tx.Status == transaction.TxStatusInvalid.String() {
-			return
-		}
-		// update transaction
-		meta, serializedData = prepareTxUpdate(tx)
-	} else {
-		// insert transaction in database
-		meta = []byte(fmt.Sprintf(`{ "index" : { "_id" : "%s" } }%s`, tx.Hash, "\n"))
-		serializedData, err = json.Marshal(tx)
-		if err != nil {
-			log.Debug("indexer: marshal",
-				"error", "could not serialize transaction, will skip indexing",
-				"tx hash", tx.Hash)
-			return
-		}
-	}
-	return
-}
+	meta = []byte(fmt.Sprintf(`{"update":{"_id":"%s", "_type": "_doc"}}%s`, tx.Hash, "\n"))
 
-func prepareSerializedAccountInfo(address string, account *AccountInfo) (meta []byte, serializedData []byte) {
-	var err error
-	meta = []byte(fmt.Sprintf(`{ "index" : { "_id" : "%s" } }%s`, address, "\n"))
-	serializedData, err = json.Marshal(account)
+	marshalledTx, err := json.Marshal(tx)
 	if err != nil {
 		log.Debug("indexer: marshal",
```

Comment:
shouldn't this be an unrecoverable error? If something goes terrible wrong, the node will just move forward, no one will read the logs but then we will start wondering why the accounts balances do not match.

Reply:
refactored return parameters and added an error also. I have also refactored the components calling this in order to return errors instead of just logging